papas/simulator.py

```diff
@@ -293,8 +293,8 @@ def simulate(self, ptcs, history):
                 # to avoid numerical problems in propagation (and avoid making a particle that is not used)
                 continue
             pdebugger.info(str('Simulating {}'.format(ptc)))
-            # ptc = pfsimparticle(gen_ptc, len(self.simulated_particles))
-            self.history[ptc.uniqueid] = Node(ptc.uniqueid)
+            uniqueid = ptc.uniqueid()
+            parent = self.history.setdefault(uniqueid, Node(uniqueid))
```

Comment:
why this change?

Reply:
Originally the simparticles were the first things in the event so there were no existing History nodes. 
Now the simparticles can descend from gen_particles and so a Node can already exist in the history. The change means that if the node already exists it will be returned, if not there a new node is made.